### ISSUE TITLE
Crash on Image select

### ISSUE THREAD

**Issue by Sepero** [oxen-io/session-android#648]
![Screenshot_20210709-134352__01](https://user-images.githubusercontent.com/1575546/125035355-e7f48500-e0bb-11eb-9490-4e7929996a76.jpg)

Bug with new interface. It worked previously.

When choosing image select, Session "crashes" out of the image selection. Session doesn't crash, but returns to the chat. I'm still able to select/send images with the file select option.

Oxygen OS 10.3.5 (Android 10)
Session 1.11.2

**Comment by hjubb** [comment]
Does this still happen on the latest version?

**Comment by Flam3z** [comment]
this happens everytime I press the gallery button. Another issue was made related to this [https://github.com/oxen-io/session-android/issues/609](url)

**Comment by hjubb** [comment]
Will look into it

**Comment by Sepero** [comment]
Possibly duplicate #609 

Still happening Session v1.11.4

**Comment by Flam3z** [comment]
Yeah it is a dupe and its still an ongoing bug.

**Comment by Sepero** [comment]
I think I read somebody mentioning having a lot of pictures in their gallery. I don't know if that's related or not. I have thousands of pictures in my gallery.

**Comment by hjubb** [comment]
That's helpful info @Sepero I think I've seen a related crash to something like this. Was it just in the root select folder or was it when you went into the "All Media" or other such folder?

**Comment by Sepero** [comment]
https://user-images.githubusercontent.com/1575546/126250249-2c3c22bc-de08-42ae-9df9-75ca3255ea92.mp4

Immediately after pressing the attach image button (image at top), the screen will go black for a second and crash out of the chat and back to chat selection screen. Attached video.

If I do it several times, Android will begin to popup the message "App crashed: show info or close"

**Comment by hjubb** [comment]
Hey @Sepero is there any chance you have the ability to capture the logcat output for this crash? Might be a challenge if you're not familiar but it'd greatly help actually sort this out, I've located only one potential this is happening for which I can't reproduce but can fix by adding null checks in the right places but it'd be great to know whether this is actually the same issue that's happening to you and being able to confirm the fix will work before diffing the same areas of code between the old conversation screen and the new one if it's only a recent issue

**Comment by Sepero** [comment]
I captured this with Scoop

```
FATAL EXCEPTION: AsyncTask #2
Process: network.loki.messenger, PID: 14309
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.String.toLowerCase()' on a null object reference
	at org.thoughtcrime.securesms.mediasend.MediaRepository.lambda$getFolders$4(MediaRepository.java:85)
	at org.thoughtcrime.securesms.mediasend.-$$Lambda$MediaRepository$rb0WGDEhH59gBRwRHH9xVcZjnSw.compare(Unknown Source:4)
	at java.util.TimSort.binarySort(TimSort.java:296)
	at java.util.TimSort.sort(TimSort.java:239)
	at java.util.Arrays.sort(Arrays.java:1492)
	at java.util.ArrayList.sort(ArrayList.java:1470)
	at java.util.Collections.sort(Collections.java:206)
	at com.annimon.stream.operator.ObjSorted.nextIteration(ObjSorted.java:25)
	at com.annimon.stream.iterator.LsaExtIterator.hasNext(LsaExtIterator.java:20)
	at com.annimon.stream.Stream.toList(Stream.java:1538)
	at org.thoughtcrime.securesms.mediasend.MediaRepository.getFolders(MediaRepository.java:86)
	at org.thoughtcrime.securesms.mediasend.MediaRepository.lambda$getFolders$0$MediaRepository(MediaRepository.java:44)
	at org.thoughtcrime.securesms.mediasend.-$$Lambda$MediaRepository$iLyJXAmiCw0RkPMFNxlE1qgJ8Bs.run(Unknown Source:6)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
```

I also recorded the logcat, but I'm not sure if I did it correctly. In the 10 seconds it took me to start saving logcat, reproduce the crash, and stop logcat, it created a 3MB file. Available below
https://cryptpad.fr/file/#/2/file/wO2KO7Pji+ijdCOHCuz93EQL/

**Comment by hjubb** [comment]
Awesome that's great looks like the exact one I was seeing happen elsewhere, it should be fixed shortly 👍 thanks so much once again for this

**Comment by Sepero** [comment]
You're welcome. I want to ask for future reference. Was the Scoop enough, or did you have to read the logcat also?

**Comment by hjubb** [comment]
The scoop was enough, pointed to the exact same issue I had seen happen elsewhere so it was good to know where it was going wrong and that it was the same problem I was already solving. I actually didn't know about scoop that seems like a really handy tool to have I'll definitely recommend it to people posting bugs that are harder to track down or reproduce in the future so thanks for that! 😊

**Comment by hjubb** [comment]
I believe this is fixed in 1.11.8, let me know if this is not the case
